### PR TITLE
python3Packages.simple-term-menu: init at 1.6.4

### PR DESCRIPTION
**Pull request by syedmoizurrehman**
## Description of changes

Add Python package [simple-term-menu](https://github.com/IngoMeyer441/simple-term-menu).

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by syedmoizurrehman** [comment]
I have tested this change by importing my local nixpkgs and using the `simple-term-menu` package. I don't see any unit tests in the upstream so, I have left `nativeCheckInputs` empty. Please let me know if this needs to be populated.

File: pkgs/development/python-modules/simple-term-menu/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  nativeBuildInputs = [ setuptools ];
+
+  disabled = pythonOlder "3.4";
+
+  pname = "simple-term-menu";
+  version = "1.6.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "be9c5dbd8df12a404b14cd8e95d6fc02d58c60e2555f65ddde41777c487fb3b9";
```
Comment (natsukium) [inline pkgs/development/python-modules/simple-term-menu/default.nix:20]:
```suggestion
    hash = "sha256-vpxdvY3xKkBLFM2Oldb8AtWMYOJVX2Xd3kF3fEh/s7k=";
```
Could you please use SRI hash instead?
https://nixos.org/manual/nixpkgs/stable/#fetchurl

File: pkgs/development/python-modules/simple-term-menu/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  };
+
+  propagatedBuildInputs = [ ];
```
Comment (natsukium) [inline pkgs/development/python-modules/simple-term-menu/default.nix:23]:
```suggestion
``` 
If the package has no dependencies, we don't need to define `propagatedBuildInputs`.

File: pkgs/development/python-modules/simple-term-menu/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  pythonImportsCheck = [ "simple_term_menu" ];
+
+  nativeCheckInputs = [ ];
```
Comment (natsukium) [inline pkgs/development/python-modules/simple-term-menu/default.nix:27]:
```suggestion
  # no unit tests in the upstream
  doCheck = false;
```

File: pkgs/development/python-modules/simple-term-menu/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+buildPythonPackage rec {
+  pyproject = true;
+  nativeBuildInputs = [ setuptools ];
+
+  disabled = pythonOlder "3.4";
+
+  pname = "simple-term-menu";
+  version = "1.6.4";
+
+  src = fetchPypi {
```
Comment (natsukium) [inline pkgs/development/python-modules/simple-term-menu/default.nix:18]:
```
  pname = ...;
  version = ...;
  pyproject = ...;
  disabled = ...;

  src = ...;

  nativeBuildInputs = ...;
  ...
```
In practice, the order does not affect the packaging itself, but it is preferable to do it like other packages to make code review and maintenance easier.

**Review by natsukium — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/simple-term-menu/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchPypi
+
```
Comment (kirillrdy) [inline pkgs/development/python-modules/simple-term-menu/default.nix:5]:
```suggestion
```

File: pkgs/development/python-modules/simple-term-menu/default.nix (hunk 4 above)
Comment (syedmoizurrehman) [reply pkgs/development/python-modules/simple-term-menu/default.nix]: Makes sense, I have updated the order.

**Comment by kirillrdy** [comment]
@GrahamcOfBorg build python310Packages.simple-term-menu python311Packages.simple-term-menu

**Review by kirillrdy — APPROVED** [review]

**Review by natsukium — APPROVED** [review]
LGTM, thanks for your contribution! Welcome to nixpkgs!🎉

**Comment by syedmoizurrehman** [comment]
@natsukium Thank you, I appreciate you and @kirillrdy taking the time to review the changes!

**Comment by natsukium** [comment]
Feel free to ping us if you would like a PR review or have any problems.

**Pull request merged**